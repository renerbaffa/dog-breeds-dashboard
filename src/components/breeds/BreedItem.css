.container {
  padding: 12px;
  font-size: 20px;
  cursor: pointer;
  border-bottom: 1px solid lightgray;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.container:last-child {
  border-bottom: none;
}

.container:hover {
  background-color: #e6e6e6;
}

.container.selected {
  background-color: #afafaf !important;
}
