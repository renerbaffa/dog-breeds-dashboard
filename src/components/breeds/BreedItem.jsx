import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import styles from './BreedItem.css';

const BreedItem = ({
  className,
  name,
  selected,
  ...other
}) => (
  <div
    {...other}
    className={
      cx(
        styles.container,
        className,
        { [styles.selected]: selected },
      )
    }
  >
    <div>{name}</div>
  </div>
);

BreedItem.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  selected: PropTypes.bool,
};

BreedItem.defaultProps = {
  className: '',
  name: '',
  selected: false,
};

export default BreedItem;
