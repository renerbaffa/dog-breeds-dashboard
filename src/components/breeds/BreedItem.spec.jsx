import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import BreedItem from './BreedItem';

describe('<BreedItem />', () => {
  let component;
  let wrapper;

  beforeEach(() => {
    component = <BreedItem name="Lhasa Apso" />;
    wrapper = shallow(component);
  });

  it('should render correctly', () => {
    const breedItem = renderer.create(component).toJSON();
    expect(breedItem).toMatchSnapshot();
  });

  it('should apply selected className when selected prop is true', () => {
    wrapper.setProps({ selected: true });
    expect(wrapper.find('.selected')).toHaveLength(1);
  });

  it('should not apply selected className when selected prop is false', () => {
    expect(wrapper.find('.selected')).toHaveLength(0);
  });
});
