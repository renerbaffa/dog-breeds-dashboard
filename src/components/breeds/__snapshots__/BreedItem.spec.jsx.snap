// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreedItem /> should render correctly 1`] = `
<div
  className="container"
>
  <div>
    Lhasa Apso
  </div>
</div>
`;
